src/hint_processor/cairo_1_hint_processor/dict_manager.rs

```diff
@@ -0,0 +1,145 @@
+    trackers: HashMap<isize, DictTrackerExecScope>,
+}
+
+impl DictTrackerExecScope {
+    /// Creates a new tracker placed in index `idx` in the dict_infos segment.
+    pub fn new(idx: usize) -> Self {
+        Self {
+            data: HashMap::default(),
+            idx,
+        }
+    }
+}
+
+impl DictManagerExecScope {
+    pub const DICT_DEFAULT_VALUE: usize = 0;
+
+    /// Allocates a new segment for a new dictionary and return the start of the segment.
+    pub fn new_default_dict(&mut self, vm: &mut VirtualMachine) -> Result<Relocatable, HintError> {
+        let dict_segment = vm.add_memory_segment();
+        if self
+            .trackers
+            .insert(
+                dict_segment.segment_index,
+                DictTrackerExecScope::new(self.trackers.len()),
+            )
+            .is_some()
+        {
+            return Err(HintError::CustomHint(String::from(
+                "Segment index already in use.",
+            )));
+        }
+
+        Ok(dict_segment)
+    }
+
+    /// Returns a reference for a dict tracker corresponding to a given pointer to a dict segment.
+    fn get_dict_tracker(&self, dict_end: Relocatable) -> Result<&DictTrackerExecScope, HintError> {
+        self.trackers
+            .get(&dict_end.segment_index)
+            .ok_or(HintError::CustomHint(String::from(
+                "The given value does not point to a known dictionary.",
+            )))
+    }
+
+    /// Returns a mut reference for a dict tracker corresponding to a given pointer to a dict
+    /// segment.
+    fn get_dict_tracker_mut(&mut self, dict_end: Relocatable) -> &mut DictTrackerExecScope {
+        self.trackers
+            .get_mut(&dict_end.segment_index)
+            .expect("The given value does not point to a known dictionary.")
+    }
+
+    /// Returns the index of the dict tracker corresponding to a given pointer to a dict segment.
+    pub fn get_dict_infos_index(&self, dict_end: Relocatable) -> Result<usize, HintError> {
+        Ok(self.get_dict_tracker(dict_end)?.idx)
+    }
+
+    /// Inserts a value to the dict tracker corresponding to a given pointer to a dict segment.
+    pub fn insert_to_tracker(&mut self, dict_end: Relocatable, key: Felt252, value: Felt252) {
+        self.get_dict_tracker_mut(dict_end).data.insert(key, value);
+    }
+
+    /// Gets a value from the dict tracker corresponding to a given pointer to a dict segment.
+    /// None if the key does not exist in the tracker data.
+    pub fn get_from_tracker(&self, dict_end: Relocatable, key: &Felt252) -> Option<Felt252> {
+        self.get_dict_tracker(dict_end).ok()?.data.get(key).cloned()
+    }
+}
+
+/// Helper object for the management of dict_squash hints.
+#[derive(Default, Debug)]
+pub struct DictSquashExecScope {
+    /// A map from key to the list of indices accessing it, each list in reverse order.
+    pub access_indices: HashMap<Felt252, Vec<Felt252>>,
+    /// Descending list of keys.
+    pub keys: Vec<Felt252>,
```

Comment:
This are probably better left private.

Reply:
pub(crate) is my best offer before we refactor the hint processor code